doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,142 @@
+r"""
+==========================================
+Interpolate images with thin-plate splines
+==========================================
+
+A conventional technique for interpolating surfaces over a set of data points
+are thin-plate splines (TPS) [1]_ [2]_.
+In an image context, given pairs of source and target control points, TPS can
+be used to transform a space, which, in our case, is a 2D image.
+
+
+.. [1] Wikipedia, Thin plate spline
+       https://en.wikipedia.org/wiki/Thin_plate_spline
+
+.. [2] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+       decomposition of deformations." IEEE Transactions on pattern analysis and
+       machine intelligence 11.6 (1989): 567–585.
+       :DOI:`10.1109/34.24792`
+       https://user.engineering.uiowa.edu/~aip/papers/bookstein-89.pdf
+
+
+Deform an image
+===============
+
+Image deformation implies displacing the pixels of an image relative to one another.
+In this example, we deform the (2D) image of an astronaut by using thin-plate splines.
+In our image, we define 6 source and target points labeled "1-6": "1-4" are found near
+the image corners, "5" near the left smile corner, and "6" in the right eye.
+At the "1-4" points, there is no displacement.
+Point "5" is displaced upward and point "6" downward.
+
+We use TPS as a very handy interpolator for image deformation.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import skimage as ski
+
+astronaut = ski.data.astronaut()
+
+src = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [240, 150], [200, 100]])
+dst = np.array([[50, 50], [400, 50], [50, 400], [400, 400], [276, 100], [230, 100]])
+
+# Fit the thin-plate spline from source (src) to target (dst) points
+
+warped_img = ski.future.tps_warp(astronaut, src[:, ::-1], dst[:, ::-1], grid_scaling=1)
```

Comment:
Passing `TpsTransform()` to `warp` already seems to work. Though, I notice that one needs to switch `src` and `dst` to receive the same results:

```python
import numpy as np
import skimage as ski

image = np.array(
    [[9, 9, 9, 9, 9, 9, 9],
     [9, 0, 0, 0, 0, 0, 9],
     [9, 0, 1, 1, 1, 0, 9],
     [9, 0, 1, 2, 1, 0, 9],
     [9, 0, 1, 1, 1, 0, 9],
     [9, 0, 0, 0, 0, 0, 9],
     [9, 9, 9, 9, 9, 9, 9]],
    dtype=float,
)
# Zoom in
src = np.array([[2, 2], [2, 4], [4, 4], [4, 2]])
dst = np.array([[1, 1], [1, 5], [5, 5], [5, 1]])

tform = ski.future.TpsTransform()
tform.estimate(src=dst, dst=src)  # switched src & dst!
using_warp = ski.transform.warp(image, tform)

using_tps_warp = ski.future.tps_warp(image, src=src, dst=dst)
np.testing.assert_array_equal(using_warp, using_tps_warp)
```

However, `tps_warp` provides `grid_scaling` as an optional performance optimization at the cost of accuracy. Would you recommend integrating that into `ski.transform.warps`? It's an optimization we took from the reference implementation...

Reply:
Also, the need to switch `src` and `dst` we took from the reference implementation
https://github.com/zpincus/celltool/blob/045920089f88a445fa2483500ef867d267ecef13/celltool/numerics/image_warp.py#L41-L43
```py
# make the reverse transform warping from the to_points to the from_points, because we
# do image interpolation in this reverse fashion
transform = _make_warp(to_points, from_points, x, y)
```

Reply:
We can make scaling a property of the ThinPlateSpline transform.